node/engine/messageservice/p2p-message-service/service.go

```diff
@@ -326,10 +274,33 @@ func (ms *P2PMessageService) addBootPeers(peers []string) {
 		peer, err := peer.AddrInfoFromP2pAddr(addr)
 		ms.checkError(err)
 
+		a, err := addressFromPeerID(peer.ID)
+		ms.checkError(err)
+		ms.peers.Store(a.String(), peer.ID)
```

Comment:
My understanding is that we can derive the SC address from a peer.ID but we cannot derive a peer.ID from a SC address. This is because an address is derived by hashing the public key and taking the last 20 bytes of that hash. This means we still need to maintain a peers map, to map SC address to peer.IDs.